fn_grpc_interface/fn_grpc_interface/components/function_grpc.py

```diff
@@ -183,55 +215,60 @@ def _function_grpc_function(self, event, *args, **kwargs):
-                logging.info("gRPC Channel Connection is not secure.")
-                # Checking Communication type
-                if _grpc_communication_type.lower().find('unary') != -1:
-                    # Initiating the connection to channel to send the data from client to grpc server
-                    try:
-                        with grpc.insecure_channel(grpc_channel) as channel:
-                            stub = grpc_stub_tuple[1](channel)
-                            stub_method = self._get_method_from_stub_object(stub, _grpc_communication_type,
-                                                                            stub_name=_grpc_rpc_stub_method_name)
-                            if grpc_function_json_data:
-                                if stub_method[1] is not None:
-                                    response_received = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data))
-
-                                    #Attempting to convert received data into JSON obejct
-                                    try:
-                                        response_received = json.dumps(response_received)
-                                        response_received = json.loads(response_received)
-                                        log.info("The Received Data is Converted into JSON Object : {}".format(response_received))
-                                    except Exception as e:
-                                        log.info("The Received data is not converted into JSON Object {}".format(e))
-                                        response_received = str(response_received)
-
-                                else:
-                                    raise FunctionError("Stub method not found, please specify the valid stub method.")
+            # Checking Communication type
+            if _grpc_communication_type.lower().find('unary') != -1:
+                try:
+                    #Connecting to gRPC Server with given Authentication type
+                    stub_class_obj,grpc_channel_obj = self._gRPC_Connect_Server(_grpc_secure_connection, grpc_channel, \
+                                                               grpc_stub_tuple[1],_grpc_certificate_path)
+                    stub_method = self._get_method_from_stub_object(stub_class_obj, _grpc_communication_type,
+                                                                    stub_name=_grpc_rpc_stub_method_name)
+                    if grpc_function_json_data:
+                        if stub_method[1] is not None:
+                            if sys.version[0] == '2':
+                                #convert unicode strings to utf-8 for python 2.7
+                                import ast
+                                grpc_function_json_data_tmp = ast.literal_eval(json.dumps(grpc_function_json_data))
+                                response_received_tmp = stub_method[1](grpc_request_tuple[1](**grpc_function_json_data_tmp))
```

Comment:
I see duplicated code between the two if branches. Maybe they can shared after the code specific to python 2.x is run.

Reply:
ok